### DISCUSSION TITLE
Can I set a value conditionally based on the existence of an optional field?

### DISCUSSION THREAD

**Discussion by NReilingh** [cuelang/cue#887] [Q&A] [answered]
I would like to define a struct that accomplishes the following:

```
#Type {
  optional?: string
  result: bool
  result: <if optional is set then true else false>
}
```

Is this possible?

**Comment by mpvl** [comment, marked as answer]
With the API or in CUE directly?

In CUE directly you can write:
```
#Type: {
	optional?: string
	result: optional != _|_
  }
  a: #Type
  a: optional: "foo"
```
Comparing against bottom is not specified in the spec, or at the very least it is poorly defined. I've written (but still need to publish) a relatively minor change, replacing this with builtins like `isdefined(optional)`. But for now this works.

In the API you can do a `LookupPath` and see whether the resulting value exists.

**Reply by NReilingh** [reply]
In CUE directly, yes -- this works! I almost got there on my own, but was comparing with `_` instead of bottom. Thanks!

**Comment by cueckoo** [comment]
This discussion has been migrated to https://github.com/cue-lang/cue/issues/887.

For more details about CUE's migration to a new home, please see https://github.com/cue-lang/cue/issues/1078.
